lib/elixir/pages/meta-programming/quote-and-unquote.md

```diff
@@ -0,0 +1,154 @@
+
+```elixir
+iex> number = 13
+iex> Macro.to_string(quote do: 11 + number)
+"11 + number"
+```
+
+That's not what we wanted, since the value of the `number` variable has not been injected and `number` has been quoted in the expression. In order to inject the *value* of the `number` variable, `unquote` has to be used inside the quoted representation:
+
+```elixir
+iex> number = 13
+iex> Macro.to_string(quote do: 11 + unquote(number))
+"11 + 13"
+```
+
+`unquote` can even be used to inject function names:
+
+```elixir
+iex> fun = :hello
+iex> Macro.to_string(quote do: unquote(fun)(:world))
+"hello(:world)"
+```
+
+In some cases, it may be necessary to inject many values inside a list. For example, imagine you have a list containing `[1, 2, 6]` and we want to inject `[3, 4, 5]` into it. Using `unquote` won't yield the desired result:
+
+```elixir
+iex> inner = [3, 4, 5]
+iex> Macro.to_string(quote do: [1, 2, unquote(inner), 6])
+"[1, 2, [3, 4, 5], 6]"
+```
+
+That's when `unquote_splicing` becomes handy:
+
+```elixir
+iex> inner = [3, 4, 5]
+iex> Macro.to_string(quote do: [1, 2, unquote_splicing(inner), 6])
+"[1, 2, 3, 4, 5, 6]"
+```
+
+Unquoting is very useful when working with macros. When writing macros, developers are able to receive code chunks and inject them inside other code chunks, which can be used to transform code or write code that generates code during compilation.
+
+## Escaping
+
+As we saw at the beginning of this chapter, only some values are valid quoted expressions in Elixir. For example, a map is not a valid quoted expression. Neither is a tuple with four elements. However, such values *can* be expressed as a quoted expression:
+
+```elixir
+iex> quote do: %{1 => 2}
+{:%{}, [], [{1, 2}]}
+```
+
+In some cases, you may need to inject such *values* into *quoted expressions*. To do that, we need to first escape those values into quoted expressions with the help of `Macro.escape/1`:
+
+```elixir
+iex> map = %{hello: :world}
+iex> Macro.escape(map)
+{:%{}, [], [hello: :world]}
+```
+
+Macros receive quoted expressions and must return quoted expressions. However, sometimes during the execution of a macro, you may need to work with values and making a distinction between values and quoted expressions will be required.
+
+In other words, it is important to make a distinction between a regular Elixir value (like a list, a map, a process, a reference, etc) and a quoted expression. Some values, such as integers, atoms, and strings, have a quoted expression equal to the value itself. Other values, like maps, need to be explicitly converted. Finally, values like functions and references cannot be converted to a quoted expression at all.
+
+You can read more about `quote` and `unquote` in the [`Kernel.SpecialForms` module](https://hexdocs.pm/elixir/Kernel.SpecialForms.html). Documentation for `Macro.escape/1` and other functions related to quoted expressions can be found in the [`Macro` module](https://hexdocs.pm/elixir/Macro.html).
```

Comment:
We no longer need to write Markdown links. :) The benefit of making it part of the documentation is that auto-linking will just work, so we can write this:

```suggestion
You can read more about `quote/1` and `unquote/1` in the `Kernel.SpecialForms` module. Documentation for `Macro.escape/1` and other functions related to quoted expressions can be found in the `Macro` module.
```

Can you please do a double pass in the docs and make sure to remove all hexdocs links in favor of just using markdown code syntax?

Reply:
Yes sure, as I said in the PR I wanted to do this later on so that we can review it in isolation but I can do that here too 👍

Reply:
Ah, sorry. I have reviewed the contents themselves several times over the years, so I am ready to review the contents too. For the other main guide, I want to port it myself, because I want to rewrite some larger sections. :)